### PR TITLE
#175449628 Build CRUD APIs for Ethnic Groups

### PR DESCRIPTION
**Pull request by Fiyin-Anne**
## What does this PR do ##
-  enables a user to get an ethnic group by id 
-  enables a user to get all ethnic groups
- enables admin to edit an existing ethnic group in the database
- enables admin to delete an ethnic group from the database

### Description of tasks to be completed? ###
- create a route to get an ethnic group by id
- create a route to get multiple ethnic groups
- create a route to delete an ethnic group by id
- create a route to edit an ethnic group
- create a controller to get an ethnic group by id
- create a controller to get multiple ethnic groups
- create a controller to delete an ethnic group by id
- create a controller to edit an ethnic group
- write tests for the controllers and routes

### How should this be manually tested/checked? ###
- Clone the PR
- Checkout to the branch `ft-get-edit-delete-cultures-175449628`
- Install the dependencies using `npm install`
- Create a `.env file` and define the following as follows:
   - `DEV_DATABASE_URL = <development database url>`
   - `TEST_DATABASE_URL = <test database url>`
   - `DATABASE_URL = <production database url>`
- run ```npm run migrate``` to create the database tables
- run ```npm run seed``` to seed the database
- Start the server by running ```npm run start-dev```
- Sign in via `http://localhost:3000/api/v1/users/signin` using a `POST` method with
```
          {
          "email: "ufuoma@gmail.com",
          "password": 12345
           }
```
and copy the token (without the quotes) from the response body.
**NOTE:** To create, update, or delete an ethnic group, set the `Authorization` to `Bearer Token` and paste the copied token in the space provided.

- To **create** a new ethnic group, make a `POST` request to `http://127.0.0.1:3000/api/v1/admin/ethnic-group/6003fb36-5112-463e-a1f9-c8944e72412f` and add the following to the body:
```
{
            "name": "Igbo",
            "festivals": "New Yam Festival",
            "dressing": "In the old days, some Igbo women celebrated their feast in ekwerike clothing. In other words, they tied thick fabric around their waist. Others preferred painting their bodies with ufle, nzu, edo, uri and other colored ornaments (instead of wearing clothing) and decorating their waist with bead accessories.",
            "language": "Igbo",
            "gallery": "https://unsplash.com/photos/eS_aZA5S42Y",
            "culturalPractices": "The Igbo people have a traditional religious belief that there is one creator, called ‘Chineke’ or ‘Chukwu’. The creator can be approached through many other deities and spirits in the form of natural objects, most commonly through the god of thunder called ‘Amadioha’."
}
```
- To **get** all ethnic groups, make a `GET` request to `http://localhost:3000/api/v1/ethnic-groups` to get all ethnic groups. Copy any ethnic group's id from the response data labeled as `id`
- To **get** a specific ethnic group, make a `GET` request to `http://localhost:3000/api/v1/ethnic-group/<ethnicgroupId>`.
- To **update** an ethnic group, make a `PATCH` request to `http://127.0.0.1:3000/api/v1/admin/ethnic-groups/<ethnicgroupId>` and place the fields to be updated in the request body. 
    - request body sample:
```
{
    "name": "Ibo"
}
```
- To **delete** an ethnic group, make a `DELETE` request to `http://127.0.0.1:3000/api/v1/admin/ethnic-group/<ethnicgroupId>` to delete the selected tourist center.

### Link to architecture
[Flow chart](https://viewer.diagrams.net/?highlight=0000ff&edit=_blank&layers=1&nav=1&title=Ethnic%20Group%20Flow%20Chart.drawio#Uhttps%3A%2F%2Fdrive.google.com%2Fuc%3Fid%3D1BukNHzgru8FGxr4yL261fbbzBheuOpsL%26export%3Ddownload)

### What are the relevant pivotal tracker stories?
[ #175449628](https://www.pivotaltracker.com/story/show/175449628)

### CODE REVIEW

**Review by balldash — DISMISSED** [review]

**Review by bellogo — APPROVED** [review]

**Review by Extremophile-ai — APPROVED** [review]

File: src/controllers/ethnicGroup.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  /**
+   *  Get all Ethnic Groups
+   *
+   * @param {object} req - The user request object
+   * @param {object} res - The user response object
+   * @returns {object} Success message
+   */
+  static async getAll(req, res) {
```
Comment (codeliezel) [inline src/controllers/ethnicGroup.js:51]: what is `getAll`? Please put proper names for your methods

File: src/controllers/ethnicGroup.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  /**
+   *  Delete an ethnic group by id
+   *
+   * @param {object} req - The user request object
+   * @param {object} res - The user response object
+   * @returns {object} Success message
+   */
+  static async getById(req, res) {
```
Comment (codeliezel) [inline src/controllers/ethnicGroup.js:68]: Please put proper names for your methods

File: src/controllers/ethnicGroup.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  /**
+   *  Delete an ethnic group by id
+   *
+   * @param {object} req - The user request object
+   * @param {object} res - The user response object
+   * @returns {object} Success message
+   */
+  static async deleteById(req, res) {
```
Comment (codeliezel) [inline src/controllers/ethnicGroup.js:127]: Please put proper names for your methods

File: src/controllers/ethnicGroup.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  /**
+   *  Update an ethnic group by id
+   *
+   * @param {object} req - The user request object
+   * @param {object} res - The user response object
+   * @returns {object} Success message
+   */
+  static async updateById(req, res) {
```
Comment (codeliezel) [inline src/controllers/ethnicGroup.js:88]: Please put proper names for your methods

File: src/controllers/ethnicGroup.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+   * @returns {object} Success message
+   */
+  static async createEthnicGroup(req, res) {
+    try {
+      const { countryId } = req.params;
+      const {
+        name, festivals, dressing, language, gallery, culturalPractices
+      } = req.body;
+      let ethnicGroupName;
+      if (name) {
```
Comment (codeliezel) [inline src/controllers/ethnicGroup.js:25]: we do not need this body of code
```
  if (name) {
        ethnicGroupName = String(name)[0].toUpperCase() + String(name).slice(1).toLowerCase();
      }
``

**Review by codeliezel — CHANGES_REQUESTED** [review]
Good job, work on the few reviews.

**Comment by codeliezel** [comment]
@Fiyin-Anne also it seems this is not in use `getByCountryId`

**Comment by Fiyin-Anne** [comment]
> @Fiyin-Anne also it seems this is not in use `getByCountryId`

No, it's not

**Review by bellogo — APPROVED** [review]

**Review by Extremophile-ai — APPROVED** [review]

**Review by codeliezel — APPROVED** [review]

**Pull request merged**